pennylane/_qutrit_device.py

```diff
@@ -0,0 +1,390 @@
+
+            elif obs.return_type is Sample:
+                results.append(self.sample(obs, shot_range=shot_range, bin_size=bin_size))
+
+            elif obs.return_type is Probability:
+                results.append(
+                    self.probability(wires=obs.wires, shot_range=shot_range, bin_size=bin_size)
+                )
+
+            elif obs.return_type is State:
+                if len(observables) > 1:
+                    raise qml.QuantumFunctionError(
+                        "The state or density matrix cannot be returned in combination"
+                        " with other return types"
+                    )
+                if self.wires.labels != tuple(range(self.num_wires)):
+                    raise qml.QuantumFunctionError(
+                        "Returning the state is not supported when using custom wire labels"
+                    )
+                # Check if the state is accessible and decide to return the state or the density
+                # matrix.
+                results.append(self.access_state(wires=obs.wires))
+
+            elif obs.return_type is not None:
+                raise qml.QuantumFunctionError(
+                    f"Unsupported return type specified for observable {obs.name}"
+                )
+
+            # TODO: Add support for MutualInfo and VnEntropy return types. Currently, qml.math is hard coded
+            # to calculate these quantities for qubit states, so it needs to be updated before these return types
+            # can be supported for qutrits.
+
+        return results
+
+    def generate_samples(self):
+        r"""Returns the computational basis samples generated for all wires.
+
+        Note that PennyLane uses the convention :math:`|q_0,q_1,\dots,q_{N-1}\rangle` where
+        :math:`q_0` is the most significant trit.
+
+        .. warning::
+
+            This method should be overwritten on devices that
+            generate their own computational basis samples, with the resulting
+            computational basis samples stored as ``self._samples``.
+
+        Returns:
+             array[complex]: array of samples in the shape ``(dev.shots, dev.num_wires)``
+        """
+        number_of_states = 3**self.num_wires
+
+        rotated_prob = self.analytic_probability()
+
+        samples = self.sample_basis_states(number_of_states, rotated_prob)
+        return QutritDevice.states_to_ternary(samples, self.num_wires)
+
+    @staticmethod
+    def generate_basis_states(num_wires, dtype=np.uint32):
+        """
+        Generates basis states in ternary representation according to the number
+        of wires specified.
+
+        Args:
+            num_wires (int): the number wires
+            dtype=np.uint32 (type): the data type of the arrays to use
+
+        Returns:
+            array[int]: the sampled basis states
+        """
+        # A slower, but less memory intensive method
```

Comment:
```suggestion
```
Comment not needed without the context of the faster method that is used in the qubit case.